node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -534,64 +536,103 @@ pub fn get_block_chain(
 }
 
 /// Get block by hash
+///
+/// - First argument is the hash of the block that we are querying.
+/// - Second argument is whether we want the response to contain a list of hashes of the
+///   transactions found in the block.  
 /* test
-{"jsonrpc":"2.0","id":1,"method":"getBlock","params":["c0002c6b25615c0f71069f159dffddf8a0b3e529efb054402f0649e969715bdb"]}
-{"jsonrpc":"2.0","id":1,"method":"getBlock","params":[{"SHA256":[255,198,135,145,253,40,66,175,226,220,119,243,233,210,25,119,171,217,215,188,185,190,93,116,164,234,217,67,30,102,205,46]}]}
+{"jsonrpc":"2.0","id":1,"method":"getBlock","params":["c0002c6b25615c0f71069f159dffddf8a0b3e529efb054402f0649e969715bdb", false]}
+{"jsonrpc":"2.0","id":1,"method":"getBlock","params":[{"SHA256":[255,198,135,145,253,40,66,175,226,220,119,243,233,210,25,119,171,217,215,188,185,190,93,116,164,234,217,67,30,102,205,46]}, false]}
 */
-pub fn get_block(hash: Result<(Hash,), jsonrpc_core::Error>) -> JsonRpcResultAsync {
-    let hash = match hash {
-        Ok(x) => x.0,
-        Err(e) => return Box::new(futures::failed(e)),
+pub fn get_block(params: Params) -> JsonRpcResultAsync {
+    let (block_hash, include_txns_hashes): (Hash, bool);
+
+    // Handle parameters as an array with a first obligatory hash field plus an optional bool field
+    if let Params::Array(params) = params {
+        if let Some(Value::String(hash)) = params.get(0) {
+            block_hash = hash.parse().unwrap()
```

Comment:
This checks that `hash` is a string, but it doesn't check if that string is a valid `Hash`. Try using a match like the one that was here before:

```rust
   let hash = match hash.parse() {
        Ok(x) => x,
        Err(e) => return Box::new(futures::failed(e)),
   };
```